ansible/library/extract_log.py

```diff
@@ -159,9 +166,12 @@ def extract_latest_line_with_string(directory, filenames, start_string):
     for filename in filenames:
         target_lines.extend(extract_line(directory, filename, start_string))
 
-    sorted_target_lines = sorted(target_lines, cmp=comparator)
+    target = target_lines[0] if len(target_lines) > 0 else None
+    for line in target_lines:
+        if comparator(line, target) > 0:
+            target = line
```

Comment:
What is the difference from 
```
sorted(target_lines, cmp=comparator) 
```
?

Reply:
Sort is sorting the whole array. The best of sorting algorithm is O(N * Log(N)).

We only need the latest entry, which can be obtained by the loop in O(N).

Reply:
This is not a huge optimization by itself. I had to open the sort call to debug. And since we could get what we need faster, I figured why not :-)

Reply:
in this case you can use max() function from python which is O(N) and just one-liner

Reply:
As we discussed in person. max doesn't take cmp=... parameter and the (key= lamda) method needs to override string.